From 4a2165b3049bec0f41c1ac1654e5edbda7e9c729 Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Fri, 27 Aug 2010 12:17:46 +0200
Subject: [PATCH] Fix stat for little endian

In future we could try to setup __dev_t with correct size
then we could remove this little/big endian ifdefs.

Signed-off-by: Michal Simek <monstr@monstr.eu>
---
 sysdeps/unix/sysv/linux/microblaze/bits/stat.h |   20 ++++++++++++++++++++
 1 files changed, 20 insertions(+), 0 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/microblaze/bits/stat.h b/sysdeps/unix/sysv/linux/microblaze/bits/stat.h
index 7f2c4e3..f38e49d 100644
--- a/sysdeps/unix/sysv/linux/microblaze/bits/stat.h
+++ b/sysdeps/unix/sysv/linux/microblaze/bits/stat.h
@@ -32,8 +32,13 @@
 
 struct stat
   {
+#ifdef __MICROBLAZEEL__
+    unsigned short int __pad1;
+    __dev_t st_dev;			/* Device.  */
+#else
     __dev_t st_dev;			/* Device.  */
     unsigned short int __pad1;
+#endif
 #ifndef __USE_FILE_OFFSET64
     __ino_t st_ino;			/* File serial number.	*/
 #else
@@ -43,8 +48,13 @@ struct stat
     __nlink_t st_nlink;			/* Link count.  */
     __uid_t st_uid;			/* User ID of the file's owner.	*/
     __gid_t st_gid;			/* Group ID of the file's group.*/
+#ifdef __MICROBLAZEEL__
+    unsigned short int __pad2;
+    __dev_t st_rdev;			/* Device number, if device.  */
+#else
     __dev_t st_rdev;			/* Device number, if device.  */
     unsigned short int __pad2;
+#endif
 	unsigned long	__pad3;
 #ifndef __USE_FILE_OFFSET64
     __off_t st_size;			/* Size of file, in bytes.  */
@@ -91,16 +101,26 @@ struct stat
 #ifdef __USE_LARGEFILE64
 struct stat64
   {
+#ifdef __MICROBLAZEEL__
+    unsigned short int __pad1;
+    __dev_t st_dev;			/* Device.  */
+#else
     __dev_t st_dev;			/* Device.  */
     unsigned short int __pad1;
+#endif
 
     __ino_t __st_ino;			/* 32bit file serial number.	*/
     __mode_t st_mode;			/* File mode.  */
     __nlink_t st_nlink;			/* Link count.  */
     __uid_t st_uid;			/* User ID of the file's owner.	*/
     __gid_t st_gid;			/* Group ID of the file's group.*/
+#ifdef __MICROBLAZEEL__
+    unsigned short int __pad2;
+    __dev_t st_rdev;			/* Device number, if device.  */
+#else
     __dev_t st_rdev;			/* Device number, if device.  */
     unsigned short int __pad2;
+#endif
 	unsigned long	__pad3;
     __off64_t st_size;			/* Size of file, in bytes.  */
     __blksize_t st_blksize;		/* Optimal block size for I/O.  */
-- 
1.5.5.1

